x/fungible/keeper/evm_hooks.go

```diff
@@ -0,0 +1,57 @@
+package keeper
+
+import (
+	cosmoserrors "cosmossdk.io/errors"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	ethcommon "github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core"
+	ethtypes "github.com/ethereum/go-ethereum/core/types"
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+	"github.com/zeta-chain/zetacore/x/fungible/types"
+)
+
+var _ evmtypes.EvmHooks = EVMHooks{}
+
+type EVMHooks struct {
+	k Keeper
+}
+
+func (k Keeper) EVMHooks() EVMHooks {
+	return EVMHooks{k}
+}
+
+// PostTxProcessing is a wrapper for calling the EVM PostTxProcessing hook on the module keeper
+func (h EVMHooks) PostTxProcessing(ctx sdk.Context, _ core.Message, receipt *ethtypes.Receipt) error {
```

Comment:
Is there a reason to not use the hooks implemented by crosschain module ?

Reply:
I think it's proper to define it in `fungible` because it exclusively uses fungible domain. So from a modular point of view, we should be able to remove `crosschain` module and still have this functionality working.